Automattic/newspack-newsletters
# fix: do not show mc_campaign_id in rest

### All Submissions:






### Changes proposed in this Pull Request:

Similar to what has been experienced with Constant Contact provider, showing the `mc_campaign_id` in rest causes an issue with synchronization:

As shared on #519:
> I struggled a little bit to implement the ability to synchronize the newsletter with the provider after the draft was created (either by the post being created with a different provider or before the tokens were available). I wasn't able to identify the root of this sync issue, but I identified that it was caused by the cc_campaign_id meta being available on the rest API. At some save cycle, the meta was being deleted and the post was desynced from the provider. Removing the meta from the API (see 2b1061b) solved the issue. I figured it's not a problem, since we don't (and probably shouldn't) manage this data on the frontend.




Closes #535.

### How to test the changes in this Pull Request:

1. In the master branch, create a newsletter using the manual provider as save as draft.
2. Change your provider to Mailchimp with a valid API key
3. Edit the newsletter by making a few changes and saving the draft a couple of times
4. Observe on your Mailchimp dashboard that you have multiple duplicated drafts
4. Checkout this branch, repeat steps and verify that you only have 1 draft

### Other information:
